R/target_market_share.R

```diff
@@ -127,102 +127,62 @@ target_market_share <- function(data,
   )
 
   check_valid_columns(data, valid_columns)
-  data <- aggregate_by_loan_id(data)
 
   crucial_scenario <- c("scenario", "tmsr", "smsp")
   check_crucial_names(scenario, crucial_scenario)
   check_crucial_names(ald, "is_ultimate_owner")
   walk_(crucial_scenario, ~ check_no_value_is_missing(scenario, .x))
 
   green_or_brown <- r2dii.data::green_or_brown
+  tmsr_or_smsp <- tmsr_or_smsp()
 
-  summary_groups <- maybe_add_name_ald(
-    c("scenario", "tmsr", "smsp", "region", "scenario_source"),
-    by_company
+  data <- aggregate_by_loan_id(data)
+
+  summary_groups <- c(
+    "scenario",
+    "tmsr",
+    "smsp",
+    "region",
+    "scenario_source",
+    "name_ald"
   )
 
   data <- join_ald_scenario(data, ald, scenario, region_isos)
 
   if (weight_production) {
     data <- summarize_weighted_production(
       data,
-      !!!rlang::syms(summary_groups),
-      use_credit_limit = use_credit_limit
+      !!!rlang::syms(summary_groups)
     )
+
   } else {
     data <- summarize_unweighted_production(
       data,
-      .data$sector_ald,
-      .data$technology,
-      .data$year,
```

Comment:
these are now internally grouped in the body of the function `summarize_unweighted_production()`